js/index.js

```diff
@@ -44,28 +43,80 @@ function renderPepperoni() {
 
 function renderMushrooms() {
   // Iteration 1: set the visibility of `<section class="mushroom">`
+  document.querySelectorAll('.mushroom').forEach(oneMushroom => {
+    if (state.mushroom) {
+      oneMushroom.style.visibility = 'visible';
+    } else {
+      oneMushroom.style.visibility = 'hidden';
+    }
+  });
 }
 
 function renderGreenPeppers() {
   // Iteration 1: set the visibility of `<section class="green-pepper">`
+ document.querySelectorAll('.green-pepper').forEach(oneGreenPepper => {
+   if (state.greenpepper) {
+     oneGreenPepper.style.visibility = 'visible';
+   } else {
+     oneGreenPepper.style.visibility = 'hidden';
+   }
+ });
 }
 
 function renderWhiteSauce() {
   // Iteration 2: add/remove the class "sauce-white" of `<section class="sauce">`
+  
+  document.querySelectorAll('.sauce').forEach(SauseWhite => {
+    if (state.whiteSauce) {
+     SauseWhite.classList.add('sauce-white');
+    } else {
+      SauseWhite.classList.remove('sauce-white');
+    }
+  });
 }
 
 function renderGlutenFreeCrust() {
   // Iteration 2: add/remove the class "crust-gluten-free" of `<section class="crust">`
+ document.querySelectorAll('.crust').forEach(CrustGlutenFree => {
+   if (state.crustglutenfree) {
+      CrustGlutenFree.classList.add('crust-gluten-free');
+     } else {
+      CrustGlutenFree.classList.remove('crust-gluten-free');
+     }
+   });
 }
 
 function renderButtons() {
   // Iteration 3: add/remove the class "active" of each `<button class="btn">`
+  document.querySelectorAll('.btn').forEach(butonActive => {
+       if (this.state) {
+      butonActive.classList.add('active');
+      } else {
+        butonActive.classList.remove('active');
+      }}
+    
+    );
+    
 }
 
+
+
 function renderPrice() {
   // Iteration 4: change the HTML of `<aside class="panel price">`
+  let panelPrice = document.querySelectorAll('.panel.price')
+  let InitialPrice = basePrice;
+  panelPrice.innerHTML = ""
+  ingredients.forEach(i => {
```

Comment:
no se puede usar forEach sobre un objeto (ingredients), pero puedes usar el método "for... in":
```
for(let i in ingredients) {
  // ...
}
```